### PR TITLE
Allow the Read status bar command to copy its contents on the third keypress 

### LINKED ISSUE #1785: Make the Read status bar command copy its contents on third press like Report window title
**Reported by Ahiiron on 2011-09-13 18:50**
Something I think would be nice is the ability to have the status line copied to the clipboard when pressed thrice, similar to say window title.

### PR DESCRIPTION
**Pull request by ehollig**
### Link to issue number:
Fixes #1785

### Summary of the issue:
When pressing the command say window title thrice, the text is copied to the clipboard. This is currently not possible with the report status bar command.

### Description of how this pull request fixes the issue:
This allows for the report status bar command to copy the information to the clipboard, similar to say window title.

### Testing performed:
Opened notepad and performed the report status bar command thrice. Pasted the contents of the status bar into the file.

### Known issues with pull request:
No information was spoken when performing the command three times in Internet Explorer 11.

### Change log entry:
• changes:
• it is now possible to press the report status bar command three times to copy the status bar text to the clipboard.

### CODE REVIEW

**Comment by bhavyashah** [comment]
@ehollig Whoa, are you coding now? Anyways, I would argue that this is
very much a user-facing feature addition and thus should be documented
in the What's New document.

On 9/8/17, Ethan Holliger <notifications@github.com> wrote:
> ### Link to issue number:
> Fixes #1785
>
> ### Summary of the issue:
> When pressing the command say window title thrice, the text is copied to the
> clipboard. This is currently not possible with the report status bar
> command.
>
> ### Description of how this pull request fixes the issue:
> This allows for the report status bar command to copy the information to the
> clipboard, similar to say window title.
>
> ### Testing performed:
> Opened notepad and performed the report status bar command thrice. Pasted
> the contents of the status bar into the file.
>
> ### Known issues with pull request:
> No information was spoken when performing the command three times in
> Internet Explorer 11.
>
> ### Change log entry:
> None needed.
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/nvaccess/nvda/pull/7577
>
> -- Commit Summary --
>
>   * Allow read status bar command to copy contents on third press to
> clipboard
>   * Allow the Read status bar command copy its contents on third press
>
> -- File Changes --
>
>     M source/globalCommands.py (9)
>
> -- Patch Links --
>
> https://github.com/nvaccess/nvda/pull/7577.patch
> https://github.com/nvaccess/nvda/pull/7577.diff
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/pull/7577
>


-- 
Best Regards
Bhavya Shah

Blogger at Hiking Across Horizons: https://bhavyashah125.wordpress.com/

Contacting Me
E-mail Address: bhavya.shah125@gmail.com
Follow me on Twitter @BhavyaShah125 or www.twitter.com/BhavyaShah125
Mobile Number: +91 7506221750

**Comment by josephsl** [comment]
Hi, as well as recorded in the user guide. CC @feerrenrut

File: source/globalCommands.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: UTF-8 -*-
+﻿# -*- coding: UTF-8 -*-
```
Comment (leonardder) [inline source/globalCommands.py]: Somehow, a weird character ended up at the start of this line. Could you have saved the file as UTF-8 with BOM? You might as well add your name to the copyright line in the process.

**Comment by leonardder** [comment]
Congratulations with your first code contribution!

**Comment by derekriemer** [comment]
Suggested whats new entry:

- changes:
- it is now possible to press the report status bar command three times to copy the status bar text to the clipboard.

**Comment by derekriemer** [comment]
Oh, by the way, congrats on your first code contribution!

File: source/globalCommands.py [hunk 2]
```diff
@@ -1361,10 +1361,13 @@ def script_reportStatusLine(self,gesture):
 			return
 		if scriptHandler.getLastScriptRepeatCount()==0:
 			ui.message(text)
-		else:
+		elif scriptHandler.getLastScriptRepeatCount()==1:
 			speech.speakSpelling(text)
+		else:
+			if api.copyToClip(text):
+				ui.message(_("%s copied to clipboard")%text)
```
Comment (derekriemer) [inline source/globalCommands.py]: You need a translators comment above this line. Something like
#Translators: The message presented when the status bar is copied to the clipboard.

File: user_docs/en/userGuide.t2t [hunk 3]
```diff
@@ -1,4 +1,4 @@
-NVDA NVDA_VERSION User Guide
+﻿NVDA NVDA_VERSION User Guide
```
Comment (derekriemer) [inline user_docs/en/userGuide.t2t]: unintentional line change? maybe crlf?

**Comment by derekriemer** [comment]
Nice work. Just one thing I found, which are required.

**Review by derekriemer — CHANGES_REQUESTED** [review]

**Review by derekriemer — APPROVED** [review]
nice fox

**Comment by derekriemer** [comment]
rar, I'm dumb. 
Nice fix

**Review by feerrenrut — APPROVED** [review]
Looks good @ehollig, thanks for the contribution! Don't worry about it for this PR, we tend to update the `en/changes.t2t` file directly on `master` when we complete the merge. We do squash merges to `master`, and regular merges to `next` during incubation. This can cause merge conflicts in the changes file. That said, its really handy that the change log entry is provided on the PR (there is a section for it in the PR template). Thanks again!

**Comment by derekriemer** [comment]
@feerrenrut Ethan isn't in contributers.txt, FYR

**Comment by dan1982code** [comment]
I noticed that the NVDA insert+1 help message for this command talks about copying the title (not status bar) to the clipboard on third press.

**Comment by josephsl** [comment]
Hi, yep, confirmed.

**Comment by ehollig** [comment]
Thank you for pointing this out. It has been corrected. CC @@feerrenrut

**Review by feerrenrut — APPROVED** [review]

**Comment by leonardder** [comment]
The grammar for the changes entry in the what's new is wrong. "it's" should be "its", and I belief it should be either "its content is", or "its contents are". I'm no expert though.

**Comment by dan1982code** [comment]
Found another typo in the changes: "compaired with installs of previous NVDA versions."

**Comment by josephsl** [comment]
Hi, I’ll address documentation issues by end of this month once most of the features are introduced. Thanks.

**Pull request merged**